src/patternfly/components/Button/button.scss

```diff
@@ -179,6 +188,19 @@
     --pf-c-button--FontSize: var(--pf-c-button--m-small--FontSize);
   }
 
+  &.pf-m-primary,
+  &.pf-m-secondary,
+  &.pf-m-tertiary,
+  &.pf-m-link:not(.pf-m-inline) {
```

Comment:
I don't think you need the `:not()` here. `.pf-m-inline` applies `padding: 0` so the vars won't apply anyways. That will keep the specificity of `.pf-m-link.pf-m-cta` the same as the others, and then you could remove `--pf-c-button--m-link--m-cta--FontWeight`, too.